### ISSUE TITLE
Transfer Learning

### ISSUE THREAD

**Issue by hoangkhoiLE** [facebookresearch/deit#17]
Can you add a tutorial about how to do a transfer learning ? Thank for your excellent work !

**Comment by TouvronHugo** [comment]
Hi @hoangkhoiLE,
Thank you for your suggestion. The code for transfer learning is the same code as for finetuning that we will add in our next repo update. We will try to make a colab notebook to make it easier to use.
Best,
Hugo

**Comment by turicumoperarius** [comment]
Hi @TouvronHugo

Is there an update available about the transfered learning colab notebook. I played around today with the awesome DEIT code and couldn't find any concrete information, besides this entry, about how to perform transfered learning with a custom dataset. I did some tests with the pytorch hymenoptera (bees and ants) dataset and also did run in the #86  "ZeroDevisionsError" problem already mentioned. I changed the amount of classes in datasets.py from 1000 to 2 but i'm still getting the same error. 

I would be really happy about a little guidance about how to do transfered learning with DEIT on custom datasets. 

Thanks

**Comment by VavoTK** [comment]
Hi @turicumoperarius  

Hi, did you manage to fine-tune / transfer learn a new model? Also can I ask which model were you using as a base model? 

Did you manually add another softmax layer on top of the model or just removed the final head and added your own? Or it was just possible to do - on the go?

**Comment by turicumoperarius** [comment]
Hi @VavoTK 

I was so far not abel to fine-tune / transfer a new DEIT model on a custom dataset. Because i had to have a quick test result from a visual transformer on a custom dataset i moved to another simpler implementation (vit-transformer pytorch) for testing. So far I did not play around any further with DEIT.

I would still be very interested to use DEIT on a custom dataset. So a little guidance on fine-tuning it would still be heavily appreciated.

**Comment by VavoTK** [comment]
Hey again, @turicumoperarius 

I actually managed to fine-tune a DeiT model. The division by 0 error occurs, because of the custom `RASampler` in the [samplers](https://github.com/facebookresearch/deit/blob/main/samplers.py) file, on the line 32

`self.num_selected_samples = int(math.floor(len(self.dataset) // 256 * 256 / self.num_replicas))`  When your dataset is smaller than 256 this just defaults to 0. Either use the default torch sampler by adding `--no-repeated-aug` in the provided command line, create a larger dataset or just remove these lines 
```if args.repeated_aug:
            sampler_train = RASampler(
                dataset_train, num_replicas=num_tasks, rank=global_rank, shuffle=True
            )
``` 

from the main.py

**Comment by rcmanga** [comment]
Thanks VavoTK. This made my day ^^

**Comment by turicumoperarius** [comment]
Thank for the hint VavoTK. I was now able to train, more or less successfull a DeiT model.
